### ISSUE TITLE
IDToken.validate_exp raises IDTokenError when exp is in the future

### ISSUE THREAD

**Issue by Avantol13** [lepture/authlib#17]
Doesn't the logic below need to flipped so that it only raises an IDTokenError when the `exp` is less than now (i.e. when the token expiration time is before now)?

```
    def validate_exp(self, now):
        if 'exp' not in self.token:
            raise IDTokenError('exp is required')
        if now and self.exp > now:  # <--- Shouldn't this be <= ?
            raise IDTokenError('exp is expired')
```

From OIDC 1.0:

> exp
> REQUIRED. Expiration time on or after which the ID Token MUST NOT be accepted for processing.

**Comment by lepture** [comment]
@Avantol13 Thanks. This is a silly mistake. I've fixed it in master branch, it will be released together with v0.4.

Currently, there are not many test cases for OIDC. This method is not covered by tests.

**Comment by Avantol13** [comment]
Gotcha, yeah I figured the OIDC stuff is still in its infancy. Just wanted to make sure it was brought to your attention
